source/Editor.js

```diff
@@ -1993,11 +1993,49 @@ var keyHandlers = {
             // If the commonAncestor was in a mathjax equation, delete the parent. Namely the mathjax equation.
             if (range.commonAncestorContainer && 
                 range.commonAncestorContainer.parentElement && 
-                checkInSVG.test(range.commonAncestorContainer.nodeName)
+                (checkInSVG.test(range.commonAncestorContainer.nodeName) || checkInSVG.test(range.commonAncestorContainer.parentElement.nodeName))
```

Comment:
The other way of expressing it might be

````javascript
var mathjaxes = self._doc.querySelectorAll('span.mathjax');
var mathjaxParent = _.find(mathjaxes, function(mathjaxEl) {
    return mathjaxEl.contains(range.commonAncestorContainer);
});


if (mathjaxParent) {
    event.preventDefault();
    detach(mathjaxParent);
}
````

Excercise for the reader: do this without underscore :(